fetcher.js

```diff
@@ -0,0 +1,71 @@
+import { url } from './url.js'
+
+async function _fetch(method, link, opts, data) {
+  // Unlike fetchival, don't silently ignore and override
+  if (opts.body) throw new Error('unexpected pre-set body option')
+
+  // Unlike fetchival, don't pollute the opts object we were given
+  const res = await fetch(link, {
+    ...opts,
+    method,
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+      ...opts.headers,
+    },
+    ...(data ? { body: JSON.stringify(data) } : {}),
+  })
+
+  if (res.status >= 200 && res.status < 300) {
+    if (opts.responseAs === 'response') return res
+    if (res.status === 204) return null
+    if (opts.responseAs === 'text') return res.text()
+    return res.json()
+  }
+
+  const err = new Error(res.statusText)
```

Comment:
```suggestion
    const err = new Error(`${res.status} ${res.statusText || ''}`.trim())
```

Fix the statusText is undefined case